libraries/AP_NavEKF2/AP_NavEKF2_Measurements.cpp

```diff
@@ -374,6 +374,19 @@ bool NavEKF2_core::readDeltaVelocity(uint8_t ins_index, Vector3f &dVel, float &d
 *             Global Position Measurement               *
 ********************************************************/
 
+/*
+  propogate forward the position data in a GPS measurement to the
+  given time.
+ */
+void NavEKF2_core::propogateGpsData(gps_elements &gpsData, uint32_t new_time_ms)
+{
+    float dtime = (new_time_ms - gpsData.time_ms) * 1.0e-3f;
+    gpsData.pos.x += dtime * gpsData.vel.x;
```

Comment:
the EKF wasn't coping with it fully. I spent a lot of time discussing this with Paul.